src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,47 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter DATE_TIME_FORMATTER
+            = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private static final int ZERO = 0;
+    private static final int ONE = 1;
+    private static final int TWO = 2;
+    private static final int THREE = 3;
+    private static final String SPLIT_BY_SPACE = " ";
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder builder = new StringBuilder("Report for period " + dateFrom + " - " + dateTo);
+        LocalDate localDateFrom = LocalDate.parse(dateFrom, DATE_TIME_FORMATTER);
+        LocalDate localDateTo = LocalDate.parse(dateTo, DATE_TIME_FORMATTER);
+        String[] dataArrays;
+        LocalDate workDay;
+        int salary;
+        int workingHours;
+        int incomePerHour;
+
+        for (int i = ZERO; i < names.length; i++) {
```

Comment:
```suggestion
        for (int i = 0; i < names.length; i++) {
```
numbers in for loop, when checking if the value is even or odd (number % 2) is ok
ZERO is not informative in general